src/analysis/AppLauncher.ts

```diff
@@ -94,16 +107,24 @@ const ApplicationLauncher = _.flow(
           _.flatMap(async ({ name, metadata: currentMetadata }) => {
             const file = getFileName(name);
             const newMetadata = currentMetadata;
-            if (shouldCopy) {
-              // clear 'outdated' metadata (which gets populated by welder) so that new copy file does not get marked as outdated
-              newMetadata[hashedOwnerEmail] = '';
+            if (file && newMetadata) {
+              if (shouldCopy) {
+                // clear 'outdated' metadata (which gets populated by welder) so that new copy file does not get marked as outdated
```

Comment:
Hmm... the issue there is this assumption that Qi wrote:

```
This happens because UI should’ve called localize API during runtime creation ([welder/server/src/main/resources/api-docs.yaml at master · DataBiosphere/welder](https://github.com/DataBiosphere/welder/blob/master/server/src/main/resources/api-docs.yaml#L89), we do something similar for Jupyter in UI…so can probably borrow some existing code), which will populate generation in welder’s cache properly.
```

We only call localize when the user actually opens the app (and for rstudio, we don't ever call localize actually, we call storage links because you launch rstudio outside of the context of a specific file unlike jupyter). The same for Jupyter, we only call localize when the user opens the file. It was never part of the original requirements specified in the UI to interact with welder as soon as the runtime is created

Reply:
it seems like welder should be responsible for things on runtime creation and the UI should be responsible for user-triggered actions (since 'runtime being created' is a stateful thing)

Reply:
Thanks! I will update the ticket